tree.go

```diff
@@ -110,10 +112,33 @@ func addDumpStatsToTree(tree treeprint.Tree, dumpStats *stats_pb.DumpStatsEntry)
 	statsTree.AddBranch(fmt.Sprintf("Pages written: %d us", dumpStats.GetPagesWritten()))
 }
 
-func addPsTreeToTree(tree treeprint.Tree, psTree *crit.PsTree) {
-	// processNodes is a recursive function to create
-	// a new branch for each process and add its child
-	// processes as child nodes of the branch.
+func addPsTreeToTree(tree treeprint.Tree, psTree *crit.PsTree) error {
+	psRoot := psTree
+	if pID != 0 {
+		// dfs performs a short-circuiting depth-first search.
+		var dfs func(*crit.PsTree) *crit.PsTree
+		dfs = func(root *crit.PsTree) *crit.PsTree {
+			if root.PID == pID {
+				return root
+			}
+			for _, child := range root.Children {
+				if ps := dfs(child); ps != nil {
+					return ps
+				}
+			}
+			return nil
+		}
+
+		ps := dfs(psTree)
+		if ps == nil {
+			return fmt.Errorf("no process with PID %d", pID)
+		}
+		psRoot = ps
+	}
+
+	/* processNodes is a recursive function to create
+	a new branch for each process and add its child
+	processes as child nodes of the branch. */
```

Comment:
For consistency, it might be better to use C++-style comment here (`//`).

Reply:
My bad, I carried the change over from the previous PR. This wasn't supposed to be there.